### PR TITLE
Fix page builder save on close route to route back to the view page

### LINKED ISSUE #552: Page builder does not remember navigation state
**Describe the bug**
When pressing save and close in page builder it always brings you to the admin panel

**To Reproduce**
Edit page, press save and close.

**Expected behavior**
Should navigate you to where you came from (e.g. page detail page)

**Screenshots**

https://user-images.githubusercontent.com/19151369/193866281-a054c04d-2ac0-402b-aaaf-712eef2d8040.mp4

### PR DESCRIPTION
**Pull request by kelanik8**
# The following changes are implemented
TODO: Summary

# Changes in the user interface:
TODO: Add screenshots, recordings or remove this section

# Checklist when submitting a final (!draft) PR
 - [ ] Commits are tidied up, squashed if needed and follow guidelines in CONTRIBUTING.md
 - [ ] Code builds
 - [ ] All existing tests pass
 - [ ] All new critical code is covered by tests
 - [ ] PR is linked to the relevant issue(s)
 - [ ] Rebased with the target branch

### CODE REVIEW

File: client/web/compose/src/store/ui.js [hunk 1]
```diff
@@ -35,6 +38,14 @@ export default function (ComposeAPI) {
 
       previousPages: (state) => state.previousPages,
 
+      previousPage: (state) => {
+        if (state.previousPage && state.previousPage.name) {
```
Comment (Fajfa) [inline client/web/compose/src/store/ui.js]: I would move this to the setPrevious page, so you dont need to check in the getter

**Review by Fajfa — CHANGES_REQUESTED** [review]

**Review by Fajfa — APPROVED** [review]

**Comment by katrinDY** [comment]
I'm not redirected to the prev page after:

- opening a record in a new tab -> page builder -> save and close
- module list -> related pages -> click on edit record page/edit record list page ->  save and close
- module edit -> related pages -> click on edit record page/edit record list page of any page -> save and close
- page tree -> page builder of any page -> save and close

**Comment by katrinDY** [comment]
One more case to resolve:
- public pages -> Home -> edit page -> save and close -> I'm redirected to page tree instead of Home in public view

**Comment by katrinDY** [comment]
I found a strange edge case for modules and charts:
- chart list -> select any chart type -> add chart name -> save -> click on back and you're redirected to create view -> click on back again and you're redirected to edit view of the new chart -> click on back again and you're redirected to edit view of the new chart => the cycle never ends
- it's the same when creating a module

_the issue isn't present when creating a ns, page or reccord_

**Pull request merged**